File: common/src/test/scala/com/taskadapter/connector/common/ValueTypeResolverTest.scala
package com.taskadapter.connector.common

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class ValueTypeResolverTest extends FunSpec with Matchers {
  describe("getValueAsString") {
    it("null to empty") {
      ValueTypeResolver.getValueAsString(null) shouldBe ""
    }
    it("string to string") {
      ValueTypeResolver.getValueAsString("abc") shouldBe "abc"
    }
    it("empty Seq to empty") {
      ValueTypeResolver.getValueAsString(Seq()) shouldBe ""
    }
    it("Seq of strings to first string") {
      ValueTypeResolver.getValueAsString(Seq("a", "b")) shouldBe "a"
    }
  }
}

File: common/src/test/java/com/taskadapter/connector/common/ValueTypeResolverTest.java
package com.taskadapter.connector.common;

import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class ValueTypeResolverTest {
    @Test
    public void nullToEmpty() {
        assertThat(ValueTypeResolver.getValueAsString(null))
                .isEqualTo("");
    }

    @Test
    public void stringToString() {
        assertThat(ValueTypeResolver.getValueAsString("abc")).isEqualTo("abc");
    }

    @Test
    public void emptyListToEmpty() {
        assertThat(ValueTypeResolver.getValueAsString(List.of()))
                .isEqualTo("");
    }

    @Test
    public void listOfStringsToFirstString() {
        assertThat(ValueTypeResolver.getValueAsString(List.of("a", "b")))
                .isEqualTo("a");
    }
}
